### DISCUSSION TITLE
Time bounding opt/abc optimization

### DISCUSSION THREAD

**Discussion by j2kun** [YosysHQ/yosys#3799] [Q&A] [answered]
Hi there!

I am somewhat novice in Yosys/ABC, and the circuit I'm optimizing takes ~2h to finish when running `abc -lut 3`, and I'm wondering if there's a mechanism by which I can specify that ABC should optimize as much as it can in, say, 10 minutes and then stop.

Similarly, I'm wondering if there is a way to time-bound `opt`, in that it could loop as long as it likes, and stop after 10 minutes or the design stops changing, whichever comes first.

**Comment by nakengelhardt** [comment, marked as answer]
We don't have any time-based limits because it would lead to non-deterministic outcomes, which is undesirable in many cases. `abc` and `opt` have `-fast` options, but if you don't want to use these universally, I'm afraid you'd have to go mess with the code. For `opt` it shouldn't be too hard to add a condition in [the while loop](https://github.com/YosysHQ/yosys/blob/236e15f3b06fc4a05143399cf10b682996d5c509/passes/opt/opt.cc#L167), but for abc since that runs in an external program you'd have to see if there's any way to get the (famously inscrutable) abc code to do what you want...
